Source/WebCore/contentextensions/ContentExtensionActions.h

```diff
@@ -233,18 +233,42 @@ struct WEBCORE_EXPORT RedirectAction {
     void applyToRequest(ResourceRequest&, const URL&);
 };
 
-struct ReportIdentifierAction {
+struct ReportIdentifierAction : public ActionWithStringMetadata<ReportIdentifierAction> {
     double identifier;
 
-    ReportIdentifierAction isolatedCopy() const { return { identifier }; }
+    ReportIdentifierAction(String string)
+    : ActionWithStringMetadata<ReportIdentifierAction> { { WTFMove(string) } }
+    , identifier(0)
+    {
+    }
+
+    ReportIdentifierAction(String string, double identifier)
+    : ActionWithStringMetadata<ReportIdentifierAction> { { WTFMove(string) } }
+    , identifier(identifier)
+    {
+    }
+
+    ReportIdentifierAction isolatedCopy() const & { return { string.isolatedCopy(), identifier }; }
+    ReportIdentifierAction isolatedCopy() && { return { WTFMove(string).isolatedCopy(), identifier }; }
     friend bool operator==(const ReportIdentifierAction&, const ReportIdentifierAction&) = default;
+
     void serialize(Vector<uint8_t>& vector) const
     {
         vector.reserveCapacity(vector.size() + sizeof(identifier));
         vector.append(asByteSpan(identifier));
+
+        ActionWithStringMetadata<ReportIdentifierAction>::serialize(vector);
+    }
+
+    static ReportIdentifierAction deserialize(std::span<const uint8_t> span)
+    {
+        return { ActionWithStringMetadata<ReportIdentifierAction>::deserialize(span.subspan(sizeof(identifier))).string, reinterpretCastSpanStartTo<double>(span) };
```

Comment:
We could use the value returned by `ActionWithStringMetadata<ReportIdentifierAction>::deserialize` here instead of just pulling out the string:
```suggestion
        ReportIdentifierAction value = ActionWithStringMetadata<ReportIdentifierAction>::deserialize(span.subspan(sizeof(identifier)));
        value.identifier = reinterpretCastSpanStartTo<double>(span);
        return value;
```